src/core/trtserver.cc

```diff
@@ -626,6 +626,49 @@ TrtServerResponse::OutputData(
   return nullptr;  // Success
 }
 
+//
+// TrtServerModelIndex
+//
+// Implementation for TRTSERVER2_ModelIndex.
+//
+class TrtServerModelIndex {
+ public:
+  TrtServerModelIndex(ni::ModelRepositoryIndex model_repository_index);
+  TRTSERVER_Error* GetModelNames(
+      const char* const** models, uint64_t* models_count);
+
+ private:
+  ni::ModelRepositoryIndex model_repository_index_;
+  std::vector<const char*> indices_;
+};
+
+TRTSERVER_Error*
+TrtServerModelIndex::GetModelNames(
+    const char* const** models, uint64_t* models_count)
+{
+  indices_.clear();
```

Comment:
Don't use indices_, get the pointers to return directly from model_repository_index_.
As it is now if I call GetModelNames a second time I will invalidate all the pointers that I got back the first time I called it.